test/functional/rpc_psbt.py

```diff
@@ -637,6 +638,65 @@ def run_ca_tests(self):
         self.nodes[0].generate(1)
         self.sync_all()
 
+        # Try a multiparty blinded tx
+        # Prepare wallets and UTXOs for inputs
+        self.nodes[2].createwallet("w1")
+        w1 = self.nodes[2].get_wallet_rpc("w1")
+        self.nodes[2].createwallet("w2")
+        w2 = self.nodes[2].get_wallet_rpc("w2")
+        self.nodes[2].createwallet("w3")
+        w3 = self.nodes[2].get_wallet_rpc("w3")
+        w1_addr = w1.getaddressinfo(w1.getnewaddress())["confidential"]
+        w2_addr = w2.getaddressinfo(w2.getnewaddress())["confidential"]
+        w3_addr = w3.getaddressinfo(w3.getnewaddress())["confidential"]
+        txid1 = self.nodes[0].sendtoaddress(w1_addr, 10)
+        txid2 = self.nodes[0].sendtoaddress(w2_addr, 10)
+        txid3 = self.nodes[0].sendtoaddress(w3_addr, 10)
+        self.sync_all()
+        vout1 = find_vout_for_address(self.nodes[2], txid1, w1_addr)
+        vout2 = find_vout_for_address(self.nodes[2], txid2, w2_addr)
+        vout3 = find_vout_for_address(self.nodes[2], txid3, w3_addr)
+        self.nodes[0].generate(1)
+        self.sync_all()
+        # Make the PSBT
+        created_psbt = self.nodes[0].createpsbt(
+            [
+                {"txid": txid1, "vout": vout1},
+                {"txid": txid2, "vout": vout2},
+                {"txid": txid3, "vout": vout3},
+            ],
+            [
+                {self.get_address(True, 0): Decimal("9.999"), "blinder_index": 0},
+                {self.get_address(True, 0): Decimal("9.999"), "blinder_index": 1},
+                {self.get_address(True, 0): Decimal("9.999"), "blinder_index": 2},
+                {"fee": Decimal("0.003")}
+            ]
+        )
+        # Update all but don't blind
+        up_psbt1 = w1.walletprocesspsbt(psbt=created_psbt, sign=False)["psbt"]
+        up_psbt2 = w2.walletprocesspsbt(psbt=created_psbt, sign=False)["psbt"]
+        up_psbt3 = w3.walletprocesspsbt(psbt=created_psbt, sign=False)["psbt"]
+        # Combine updated
+        comb_psbt1 = self.nodes[0].combinepsbt([up_psbt1, up_psbt2, up_psbt3])
+        # 1 and 2 blind
+        blind_psbt1 = w1.walletprocesspsbt(psbt=comb_psbt1, sign=False)["psbt"]
```

Comment:
If you double-up this line, e.g. by writing
```
         blind_psbt1 = w1.walletprocesspsbt(psbt=blind_psbt1, sign=False)["psbt"]
```
after it, you will get a failure of the form `Transaction has blind inputs belonging to this blinder but does not have outputs to blind`.

The reason is that at the beginning of `BlindPSBT` in blindpsbt.cpp, we check "is everything done" by comparing the number of fully blinded outputs with the number of outputs -- but we do this **before** filtering out outputs that aren't ours. Then when we do the actual blinding, skipping all the outputs that (a) weren't ours, and (b) were already blinded, so we skip all of them. Finally at the bottom of the function we check whether we had blinded inputs (yes) and whether we blinded no outputs (also yes) and then throw this error.

Can you add this extra test line, perhaps with a comment explaining its purpose (it should be a no-op and not throw any errors), and change the logic in blindpsbt.cpp?

Reply:
Fixed, with test.